File: voice-telephony/sip/register-sip-device/code-samples/createSIPRegistration.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# POST BODY
body = {
    'device': [
        {
            'id': '<ENTER VALUE>',
            'appExternalId': '<ENTER VALUE>',
            'computerName': '<ENTER VALUE>'
        }
    ],
    'sipInfo': [
        {
            'transport': 'UDP'
        }
    ]
}

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])
r = platform.post('/restapi/v1.0/client-info/sip-provision', body)

File: voice-telephony/sip/register-sip-device/code-samples/createSIPRegistration.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// POST BODY
const body = {
    device: [
        {
            id: '<ENTER VALUE>',
            appExternalId: '<ENTER VALUE>',
            computerName: '<ENTER VALUE>'
        },
    ],
    sipInfo: [
        {
            transport: 'UDP'
        },
    ]
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.post(`/restapi/v1.0/client-info/sip-provision`, body).then((r) => {
        // PROCESS RESPONSE
    });
});
